### PR TITLE
[pytest] Add auto-restart related functions in devices.py

### PR DESCRIPTION
**Pull request by yozhao101**
- [ ] Bug fix
- [x] Testbed and Framework(new/improvement)
- [ ] Test case(new/improvement)

#### What is the motivation for this PR?
I added two functions `get_container_autorestart_states(...)` and `get_critical_group_and_process_list(...)` which
are related to auto-restart feature in devices.py.

#### How did you verify/test it?
I tested these two functions on virtual testbed.

### CODE REVIEW

File: tests/common/devices.py [hunk 1]
```diff
@@ -332,6 +332,36 @@ def critical_services_fully_started(self):
         logging.debug("Status of critical services: %s" % str(result))
         return all(result.values())
 
+    def get_critical_group_and_process_list(self, container_name):
```
Comment (jleveque) [inline tests/common/devices.py:335]: Function returns two lists, name should be plural to help indicate this: `get_critical_group_and_process_lists`

File: tests/common/devices.py [hunk 2]
```diff
@@ -332,6 +332,36 @@ def critical_services_fully_started(self):
...
+        @return: Two lists which include the critical groups and critical processes respectively
+        """
+        critical_group_list = []
+        critical_process_list = []
+        succeeded = True
+
+        file_content = self.shell("docker exec {} bash -c '[ -f /etc/supervisor/critical_processes ] \
+                && cat /etc/supervisor/critical_processes'".format(container_name), module_ignore_errors=True)
+        for line in file_content["stdout_lines"]:
+            line_info = line.strip('\n').split(':')
```
Comment (jleveque) [inline tests/common/devices.py:348]: There should be no trailing whitespace. You can use `line.strip()` without the `'\n'`.

**Review by jleveque — CHANGES_REQUESTED** [review]

File: tests/common/devices.py (hunk 1 above)
Comment (yozhao101) [reply tests/common/devices.py:335]: Great suggestion! Updated.

File: tests/common/devices.py (hunk 2 above)
Comment (yozhao101) [reply tests/common/devices.py]: Updated!

File: tests/common/devices.py [hunk 3]
```diff
@@ -349,12 +380,10 @@ def critical_process_status(self, service):
...
             return result
 
-        # get critical process list for the service
-        output = self.command("docker exec {} bash -c '[ -f /etc/supervisor/critical_processes ] && cat /etc/supervisor/critical_processes'".format(service), module_ignore_errors=True)
-        for l in output['stdout'].split():
-            # If ':' exists, the second field is got. Otherwise the only field is got.
-            critical_process_list.append(l.split(':')[-1].rstrip())
-        if len(critical_process_list) == 0:
+        # get critical group and process lists for the service
+        critical_group_list, critical_process_list, succeeded = self.get_critical_group_and_process_list(service)
```
Comment (jleveque) [inline tests/common/devices.py:384]: Build is failing because you updated the method name but didn't update this call to match.

**Review by jleveque — CHANGES_REQUESTED** [review]
vsimage test is failing because you updated the function

File: tests/common/devices.py (hunk 3 above)
Comment (yozhao101) [reply tests/common/devices.py:384]: Will update!

**Review by jleveque — APPROVED** [review]

**Review by yxieca — APPROVED** [review]

**Pull request merged**